idaes/core/initialization/tests/test_block_triangularization.py

```diff
@@ -0,0 +1,145 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Tests for Block Triangularization initialization
+"""
+import pytest
+import types
+
+from pyomo.environ import ConcreteModel, Constraint, value, Var
+
+from idaes.core.initialization.block_triangularization import (
+    BlockTriangularizationInitializer,
+)
+from idaes.core.initialization.initializer_base import InitializationStatus
+
+from idaes.core import FlowsheetBlock
+from idaes.models.unit_models.cstr import CSTR
+from idaes.models.properties.examples.saponification_thermo import (
+    SaponificationParameterBlock,
+)
+from idaes.models.properties.examples.saponification_reactions import (
+    SaponificationReactionParameterBlock,
+)
+
+__author__ = "Andrew Lee"
+
+
+@pytest.mark.integration
+def test_workflow():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+
+    m.fs.properties = SaponificationParameterBlock()
+    m.fs.reactions = SaponificationReactionParameterBlock(
+        property_package=m.fs.properties
+    )
+
+    m.fs.unit = CSTR(
+        property_package=m.fs.properties,
+        reaction_package=m.fs.reactions,
+        has_equilibrium_reactions=False,
+        has_heat_transfer=True,
+        has_heat_of_reaction=True,
+        has_pressure_change=True,
+    )
+
+    m.fs.unit.inlet.flow_vol.fix(1.0e-03)
+    m.fs.unit.inlet.conc_mol_comp[0, "H2O"].fix(55388.0)
+    m.fs.unit.inlet.conc_mol_comp[0, "NaOH"].fix(100.0)
+    m.fs.unit.inlet.conc_mol_comp[0, "EthylAcetate"].fix(100.0)
+    m.fs.unit.inlet.conc_mol_comp[0, "SodiumAcetate"].fix(0.0)
+    m.fs.unit.inlet.conc_mol_comp[0, "Ethanol"].fix(0.0)
+
+    m.fs.unit.inlet.temperature.fix(303.15)
+    m.fs.unit.inlet.pressure.fix(101325.0)
+
+    m.fs.unit.volume.fix(1.5e-03)
+    m.fs.unit.heat_duty.fix(0)
+    m.fs.unit.deltaP.fix(0)
+
+    initializer = BlockTriangularizationInitializer(constraint_tolerance=2e-5)
```

Comment:
1. I am going to push back on the shorter names - I think these need to be specific as to what they are - i.e. they need to say they are initializers. Importing should support auto-completion in a decent IDE and users can assign shorter aliases if they wish.
2. Because the tolerance of the internal solvers was 1e-5 (I think), and once constraint had a residual of ~1.5e-5.
3. Logger output is on the cards, and I have asked for that to be added to the Pyomo solver for us.